File: fap_online_ui/lib/features/admin/data/services/admin_profile_service.dart
import '../../../../services/api_service.dart';
import '../../../../config/api_endpoints.dart';
import '../models/admin_models.dart';

class AdminProfileService {
  final ApiService _api = ApiService();

  Future<AdminProfileModel> getProfile() async {
    final data = await _api.get(ApiEndpoints.adminProfile);
    return AdminProfileModel.fromJson(data as Map<String, dynamic>);
  }

  Future<AdminProfileModel> updateProfile(
    AdminProfileUpdateRequest request,
  ) async {
    final data = await _api.put(
      ApiEndpoints.adminProfile,
      body: request.toJson(),
    );
    return AdminProfileModel.fromJson(data as Map<String, dynamic>);
  }

  Future<void> changePassword(AdminChangePasswordRequest request) async {
    await _api.put(
      ApiEndpoints.adminChangePassword,
      body: request.toJson(),
    );
  }
}

File: fap_online_backend/fap_online_backend/src/main/java/com/example/demo/service/AdminProfileService.java
package com.example.demo.service;

import com.example.demo.dto.AdminProfileDTO;
import com.example.demo.dto.AdminProfileUpdateRequest;
import com.example.demo.dto.ChangePasswordRequest;

public interface AdminProfileService {

    AdminProfileDTO getProfile();

    AdminProfileDTO updateProfile(AdminProfileUpdateRequest request);

    void changePassword(ChangePasswordRequest request);
}
